File: Platforms/iOS/TestEngine/TestEngine/Movable.swift
//
//  Movable.swift
//  TestEngine
//
//  Created by Vladas Zakrevskis on 7/3/17.
//  Copyright © 2017 VladasZ. All rights reserved.
//

import Foundation

class Movable : Entity {
    
    var velocity:  Point = Point()
    
    func update() {
        
        let nextPosition = position + velocity
        
        if nextPosition.x < 0 || nextPosition.x > 600 {
            velocity.x = -velocity.x
            direction.x = -direction.x
        }
        if nextPosition.y < 0 || nextPosition.y > 300 {
            velocity.y = -velocity.y
            direction.y = -direction.y
        }
        
        position += velocity
    }
}

File: source/testengine/Demo/2D/GameObjects/Movable.cpp
//
//  Movable.cpp
//  TestEngine
//
//  Created by Vladas Zakrevskis on 11/03/2018.
//  Copyright © 2018 VladasZ. All rights reserved.
//

#include "Movable.hpp"


void Movable::update() {
    setPosition(_position + velocity);
}
